src/Microsoft.VisualStudio.SDK.Analyzers.Tests/VSSDK006CheckServicesExistAnalyzerTests.cs

```diff
@@ -614,6 +614,58 @@ class Test : Package {
         await Verify.VerifyAnalyzerAsync(test);
     }
 
+    [Fact]
+    public async Task FieldAssigned_GetService_ThenUsedElsewhereWithIfEqualCheckAsync()
+    {
+        var test = @"
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+
+class Test : Package {
+    IVsBuildManagerAccessor svc;
+    protected override void Initialize() {
+        base.Initialize();
+        this.svc = this.GetService(typeof(SVsBuildManagerAccessor)) as IVsBuildManagerAccessor;
+    }
+
+    void Foo() {
+        if (svc == null) {
+            return;
+        }
+        svc.BeginDesignTimeBuild();
+    }
+}
+";
+
+        await Verify.VerifyAnalyzerAsync(test);
+    }
+
+    [Fact]
+    public async Task FieldAssigned_GetService_ThenUsedElsewhereWithIfIsNullCheckAsync()
+    {
+        var test = @"
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+
+class Test : Package {
+    IVsBuildManagerAccessor svc;
+    protected override void Initialize() {
+        base.Initialize();
+        this.svc = this.GetService(typeof(SVsBuildManagerAccessor)) as IVsBuildManagerAccessor;
+    }
+
+    void Foo() {
+        if (svc is null) {
+            return;
+        }
+        svc.BeginDesignTimeBuild();
```

Comment:
Can you add a test (and support for, if necessary) the alternate syntax:

```cs
if (svc is object) {
   svc.BeginDesignTimeBuild();
}
```

also

```cs
if (svc is { }) {
   svc.BeginDesignTimeBuild();
}
```

or of course:

```cs
if (svc is IVsBuildManagerAccessor) {
   svc.BeginDesignTimeBuild();
}
```

Reply:
Done for `is object` and `is IVsBuildManagerAccessor`.
`is { }` is a C# 8 pattern match syntax.